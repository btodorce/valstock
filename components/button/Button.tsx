import { DetailedHTMLProps, FC, ButtonHTMLAttributes } from "react"
import styles from "./Button.module.scss"

type Button = Omit<
	DetailedHTMLProps<ButtonHTMLAttributes<HTMLButtonElement>, HTMLButtonElement>,
	"disabled" & "style"
>

interface P {
	loading?: boolean
}

export const Button: FC<P & Button> = ({ loading = false, children, ...props }) => {
	return (
		<button className={styles.container} disabled={loading} {...props}>
			{children}
		</button>
	)
}

Button.displayName = "Button"

export default Button
