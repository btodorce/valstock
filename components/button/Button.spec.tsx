import { render } from "@testing-library/react";
import renderer from "react-test-renderer";

import Button from "./Button";

const defaultProps = {
    loading: false
};

describe("Button", () => {
    it("should render successfully", () => {
        const tree = renderer.create(<Button {...defaultProps} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
