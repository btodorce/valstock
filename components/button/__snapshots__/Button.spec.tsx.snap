// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render successfully 1`] = `
<button
  className="container"
  disabled={false}
/>
`;
